packages/host/app/resources/room.ts

```diff
@@ -355,9 +358,13 @@ export class RoomResource extends Resource<Args> {
       return this._skillCardsCache.get(cardId);
     }
 
-    let skillCard = await this.store.add<Skill>(cardDoc, {
-      doNotPersist: true,
-    });
+    let skillCard: Skill;
+    let maybeSkillCard = await this.store.get(cardId);
+    if (isCardInstance(maybeSkillCard)) {
+      skillCard = maybeSkillCard as Skill;
+    } else {
+      skillCard = await this.store.add<Skill>(cardDoc);
+    }
```

Comment:
Looks good. ultimately we have this ticket https://linear.app/cardstack/issue/CS-8367/clarify-how-command-result-cards-that-live-in-matrix-interact-with which will update the store to be able to load instances from matrix state instead of the store. perhaps this is what we'll want in the future?

Reply:
Yeah, it would be good if the store handled this.